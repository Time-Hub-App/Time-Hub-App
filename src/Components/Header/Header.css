header h3:hover {
  cursor: pointer;
  text-decoration: underline;
}

header {
  background-color: #727397;
  color: aliceblue;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 30px;
}

.active {
  text-decoration: underline;
}
