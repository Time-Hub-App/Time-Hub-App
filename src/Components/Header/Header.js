import './Header.css';
import { logout } from '../../Services/users';
import { useHistory } from 'react-router-dom';
import classNames from 'classnames';

export default function Header({ user, setUser, home, auth, userpage }) {
  const history = useHistory();

  const handleHome = () => {
    history.push('/');
  };
  const handleAuth = () => {
    history.push('/auth');
  };
  const handleUserpage = () => {
    history.push('/journals');
  };
  const handleLogout = async () => {
    await logout();
    setUser(null);
  };

  return (
    <header>
      <h3 className={classNames({ active: home })} onClick={handleHome}>
        Home
      </h3>
      <h3 className={classNames({ active: auth })} onClick={handleAuth}>
        Log In
      </h3>
      <h3 className={classNames({ active: userpage })} onClick={handleUserpage}>
        Journals
      </h3>
      {user && <h3 onClick={handleLogout}>LogOut</h3>}
    </header>
  );
}
