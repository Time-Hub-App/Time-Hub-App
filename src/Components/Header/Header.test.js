import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from './Header';

it('should render a header', () => {
  const { container } = render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
