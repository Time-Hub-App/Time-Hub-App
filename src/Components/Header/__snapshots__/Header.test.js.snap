// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a header 1`] = `
<div>
  <header>
    <h3
      class=""
    >
      Home
    </h3>
    <h3
      class=""
    >
      About
    </h3>
    <h3
      class=""
    >
      Log In
    </h3>
    <h3
      class=""
    >
      Journals
    </h3>
  </header>
</div>
`;
